src/resolvers.ts

```diff
@@ -53,5 +53,40 @@ export const resolvers: ResolverMap = {
       console.warn(`Could not load value ${argument} from credstash: ${error}`);
       return undefined;
     });
+  },
+  aws: async (argument: string) => {
+    const params = argument.split('.');
+    const [secretKey, additonalValue] = params;
+
+    if (!secretKey) {
+      console.warn('No key provided to aws secret manager resolver');
+      return undefined;
+    }
+
+    const client = new AWS.SecretsManager();
+
+    try {
+      const { SecretString } = await client
+        .getSecretValue({ SecretId: secretKey })
+        .promise();
+
+      if (additonalValue) {
+        const parseJson = JSON.parse(SecretString);
+
+        const lookupSecretValue = parseJson[additonalValue];
+
+        if (!lookupSecretValue) {
+          console.warn(`No aws secret manager value found for ${secretKey}.${additonalValue}`);
+          return undefined;
+        }
+
+        return lookupSecretValue;
+      }
+
+      return SecretString;
+    } catch (e) {
+      console.warn(`Invalid key passed to secret manager ${secretKey}`);
```

Comment:
Is this the only exception that can be thrown here?

Reply:
Maybe there’s a way to avoid exceptions entirely here.

Reply:
These are the 2 exceptions I was able to re-produce from the `aws-sdk`. I've updated the test to reflect the ladder. I'm not sure how we would avoid them entirely though? There is always the possibility to receive an exception from the `aws-sdk`. 

```
{
  message: 'The security token included in the request is invalid.',
  code: 'UnrecognizedClientException',
  ...
}
```
```
{
  message: "Secrets Manager can't find the specified secret.",
  code: 'ResourceNotFoundException',
  ...
}
```